### PR TITLE
Added Fenwick Tree in Java and JavaScript

### LINKED ISSUE #922: Implement Fenwick Tree in Java and JavaScript
I want to add Fenwick Tree code in Two languages one Java and another JavaScript.
@jainaman224 please assign this issue.

## Files to be added
- Fenwick_Tree.java
- Fenwick_Tree.js

### PR DESCRIPTION
**Pull request by dshahid380**
Fixes #922  

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Adding Fenwick Tree in Java
- Adding Fenwick Tree in JavaScript

#### Languages Used:
- Java
- JavaScript

#### Files Added:
- Fenwick_Tree.java
- Fenwick_Tree.js

### CODE REVIEW

**Comment by dshahid380** [comment]
@SakshiShreya  I have made changes that you have proposed. Please review this.

**Comment by dshahid380** [comment]
One more thing, if you look at files changed then you will find that my codes don't have proper indentation but if you look my forked repo you will find that my codes have proper indentation. I don't know why this happening.

**Comment by jainaman224** [comment]
Please provide me the link to the file in your fork.

**Comment by dshahid380** [comment]
@jainaman224 and @SakshiShreya please see this
https://github.com/dshahid380/Algo_Ds_Notes/blob/master/Fenwick_Tree/Fenwick_Tree.java
https://github.com/dshahid380/Algo_Ds_Notes/blob/master/Fenwick_Tree/Fenwick_Tree.js

**Comment by jainaman224** [comment]
Copied from geeks for geeks. https://github.com/dshahid380/Algo_Ds_Notes/blob/master/Fenwick_Tree/Fenwick_Tree.java#L55-L59 and indentation error in geeks for geeks are exact same. It can't be coincidence.
Please try to learn not to copy.

**Comment by dshahid380** [comment]
@jainaman224 sir?

**Comment by jainaman224** [comment]
Please

**Comment by jainaman224** [comment]
Please justify the coincidence of exact indentation error. Whereas it fixed for Js code.

**Comment by dshahid380** [comment]
@jainaman224 I don't know how it occurred but you can check that i have implemented the Fenwick Tree in Python which is already merged and this time i created PR in Java and JS. If i copied the code then how could i implement it in two different languages(Python and JS).

File: Fenwick_Tree/Fenwick_Tree.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    static void build(int arr[], int size) {
+        /*
+	Argument
+	arr : The original array
+        size : The length of the given array
+        This function will construct the Fenwick Tree 
+        from the given array of length "size"
+	*/	
+
+        for(int i=0; i<size; i++) {
```
Comment (SakshiShreya) [inline Fenwick_Tree/Fenwick_Tree.java]: Add space around operators.

**Comment by dshahid380** [comment]
I don't know how it happened but if you see my commits in my repo which reflects here.
https://github.com/dshahid380/Algo_Ds_Notes/blob/master/Fenwick_Tree/Fenwick_Tree.java
https://github.com/dshahid380/Algo_Ds_Notes/blob/master/Fenwick_Tree/Fenwick_Tree.js
Please see here @SakshiShreya , you will find that the improvement you asked have already been done.

**Comment by SakshiShreya** [comment]
The space around operators in line 74 is not present even in your repo. Please fix that too.

**Comment by dshahid380** [comment]
@SakshiShreya and @jainaman224 , i am creating a fresh PR and closing this one.

**Pull request closed without merging**